src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -7,6 +7,95 @@
 
 namespace System.Text.Json
 {
+    public partial class JsonArray : System.Text.Json.JsonNode, System.Collections.Generic.ICollection<System.Text.Json.JsonNode>, System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode>, System.Collections.Generic.IList<System.Text.Json.JsonNode>, System.Collections.IEnumerable
+    {
+        public JsonArray() { }
+        public JsonArray(System.Collections.Generic.IEnumerable<bool> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<byte> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<double> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<short> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<int> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<long> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<sbyte> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<float> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<string> values) { }
+        public JsonArray(System.Collections.Generic.IEnumerable<System.Text.Json.JsonNode> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<ushort> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<uint> values) { }
+        [System.CLSCompliantAttribute(false)]
+        public JsonArray(System.Collections.Generic.IEnumerable<ulong> values) { }
+        public int Count { get { throw null; } }
+        public bool IsReadOnly { get { throw null; } }
+        public System.Text.Json.JsonNode this[int idx] { get { throw null; } set { } }
+        public void Add(bool value) { }
```

Comment:
There's a design point here around mutability vs immutability.  Make sure you have scenarios for mutability that justify the added complexity it entails.

Reply:
I am not sure if I know what you mean. `Add(JsonNode)` is anyway required by the interface, `Add(bool)` and other types are here for creating Json arrays comfortably with collection initializers and not adding a lot of `new`s. What added complexity are you thinking of?

Reply:
I was mainly considering the complexity of Add/Remove and everything else folks might want to do when they can change the size (essentially List vs Array, or IList vs ICollection).  I think collection initializers and being able to simply append an item to the array are decent scenarios that warrant making this mutable.  Make sure it can support all the sorts of things that folks might want to do with a mutable collection: look at List<T> or ArrrayList for inspiration (not necessarily copy everything).

Reply:
Ok, thanks for the explanation! I will think about additional scenarios.

Reply:
I will address it in a new PR.